index.js

```diff
@@ -179,7 +179,7 @@ Request.prototype.run = function () {
                 }
             }
 
-            if (res.status >= 400) {
+            if (!res.status || res.status < 100 || res.status >= 400) {
```

Comment:
Which status code would that be? 

I think from a user's perspective not inventing one is okay, as we normally handle only specific errors.